File: back/KaemeWebApi/Dominio/Services/PecaService.cs
using Dominio.Interface;
using Dominio.Models;
using Shared.RepositoryShared;

namespace Dominio.Services
{
    public class PecaService
    {
        private IPecaRepository _pecaRepository;

        public PecaService(IPecaRepository pecaRepository)
        {
            _pecaRepository = pecaRepository;
        }

        public RepositoryResult Create(Peca peca)
        {
            try
            {
                var result = _pecaRepository.Create(peca);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult Update(Peca peca)
        {
            try
            {
                var result = _pecaRepository.Update(peca);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult Delete(int id)
        {
            try
            {
                var result = _pecaRepository.Delete(id);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult GetTop100()
        {
            try
            {
                var result = _pecaRepository.GetTop100();
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult GetPecaById(int id)
        {
            try
            {
                var result = _pecaRepository.GetPecaById(id);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult GetPecaBySearch(string? codigo, int? tipo_peca_id, int? fornecedor_id, bool? com_estoque = false)
        {
            try
            {
                var result = _pecaRepository.GetPecaBySearch(codigo, tipo_peca_id, fornecedor_id, com_estoque);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public List<Peca> GetPecaBySearchReport(string? codigo, int? tipo_peca_id, int? fornecedor_id, bool? com_estoque = false)
        {
            var pecas = _pecaRepository.GetPecaBySearch(codigo, tipo_peca_id, fornecedor_id, com_estoque);
            return pecas;
        }        

    }
}

File: front/kaemeweb/src/app/modules/shared/services/peca.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, tap } from 'rxjs';
import { Peca } from '../models/peca';

@Injectable({
  providedIn: 'root'
})
export class PecaService {

  constructor(private httpClient: HttpClient) { }

  create(peca: Peca): Observable<boolean> {
    return this.httpClient.post<boolean>('https://localhost:7072' + '/peca/Create', peca);
  }

  update(peca: Peca): Observable<boolean> {
    return this.httpClient.put<boolean>('https://localhost:7072' + '/peca/Update', peca);
  }

  delete(id: number): Observable<boolean> {
    let params = new HttpParams()
    params = params.append('id', id)

    return this.httpClient.delete<boolean>('https://localhost:7072' + '/peca/Delete', {params});
  }

  getTop100(): Observable<Peca[]> {
    return this.httpClient.get<Peca[]>('https://localhost:7072' + '/peca/GetTop100');
  }

  getPecaById(id: number): Observable<Peca> {
    let params = new HttpParams()
    params = params.append('id', id)

    return this.httpClient.get<Peca>('https://localhost:7072' + '/peca/GetPecaById', {params});
  }

  getPecaBySearch(codigo?: string, tipo_peca_id?: number, fornecedor_id?: number, com_estoque?: boolean, ativo?: boolean): Observable<Peca[]> {
    let params = new HttpParams()
    params = params.append('codigo', codigo)
    params = params.append('tipo_peca_id', tipo_peca_id)
    params = params.append('fornecedor_id', fornecedor_id)
    params = params.append('com_estoque', com_estoque)
    params = params.append('ativo', ativo)

    return this.httpClient.get<Peca[]>('https://localhost:7072' + '/peca/GetPecaBySearch', {params});
  }

}
